src/psyclone/psyir/nodes/node.py

```diff
@@ -1023,27 +1032,34 @@ class or a tuple of such classes
             local_list += child.walk(my_type, stop_type)
         return local_list
 
-    def ancestor(self, my_type, excluding=None, include_self=False):
+    def ancestor(self, my_type, excluding=None, include_self=False,
+                 limit=None):
         '''
-        Search back up tree and check whether we have an ancestor that is
-        an instance of the supplied type. If we do then we return
+        Search back up the tree and check whether this node has an ancestor
+        that is an instance of the supplied type. If it does then we return
         it otherwise we return None. An individual (or tuple of) (sub-)
         class(es) to ignore may be provided via the `excluding` argument. If
-        include_self is True then the current node is included in the search.
+        `include_self` is True then the current node is included in the search.
+        If `limit` is provided then the search ceases if/when the supplied
+        node is encountered.
 
         :param my_type: class(es) to search for.
-        :type my_type: type or tuple of types
-        :param tuple excluding: individual (or tuple of) (sub-)class(es) to \
-                                ignore or None.
+        :type my_type: type | Tuple[type]
+        :param excluding: (sub-)class(es) to ignore or None.
+        :type excluding: Optional[type, Tuple[type]]
```

Comment:
Oops, yes.

Reply:
In fact it is `Optional[type | Tuple[type, ...]]` since "To specify a variable-length tuple of homogeneous type, use literal ellipsis" (https://docs.python.org/3/library/typing.html#typing.Tuple)

This is only change so I will do it myself before merging.